### PR TITLE
[RTM] animated tracked markers

### PR DESCRIPTION
**Pull request by thasaarah**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document [docs/contribution.md]?
* [x] Have you checked to ensure there aren't other open [Pull Requests] for the same update/change?
* [x] Have you opened/linked the issue related to your pull request?
* [x] Have you used the tag [WIP] for on-going changes, and removed it when the pull request was ready?
* [ ] When ready to merge, have you sent a comment pinging @pariterre in it?

### New Feature Submissions:

1. [ ] Does your submission pass the tests (if not please explain why this is intended)?
2. [ ] Did you write a proper documentation (docstrings and ReadMe)
3. [ ] Have you linted your code locally prior to submission (using the command: `black . -l120 --exclude "external/*"`)?

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new examples for your core changes, as applicable?
* [ ] Have you written new tests for your core changes, as applicable?

---
This change is [Reviewable](https://reviewable.io/reviews/pyomeca/bioptim/500)

related to #455

### CODE REVIEW

**Review by pariterre — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pyomeca/bioptim/500)* status: all files reviewed, 1 unresolved discussion (waiting on @thasaarah)

___
*[`bioptim/optimization/solution.py` line 1048 at r1](https://reviewable.io/reviews/pyomeca/bioptim/500#-N7W03YK0P9nSHAfPQRY:-N7W03YK0P9nSHAfPQRZ:b-dngog1) ([raw file](https://github.com/pyomeca/bioptim/blob/1cc3c2477d261f4306d6f22b67b511108928ff6d/bioptim/optimization/solution.py#L1048)):*
> ```Python
>                 if objective.type in (ObjectiveFcn.Mayer.TRACK_MARKERS, ObjectiveFcn.Lagrange.TRACK_MARKERS):
>                     if objective.node[0] in (Node.ALL, Node.ALL_SHOOTING):
>                         if objective.target is not None:
> ```

This looks like it is additive conditions (if .. AND ... AND...) 
If so, please make it explicit in a single if instead of nesting if

I think it also make sense to test (objective.target) first 



_Code quote:_
```Python
                if objective.type in (ObjectiveFcn.Mayer.TRACK_MARKERS, ObjectiveFcn.Lagrange.TRACK_MARKERS):
                    if objective.node[0] in (Node.ALL, Node.ALL_SHOOTING):
                        if objective.target is not None:
```

**Review by thasaarah — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pyomeca/bioptim/500)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @pariterre)

___
*[`bioptim/optimization/solution.py` line 1048 at r1](https://reviewable.io/reviews/pyomeca/bioptim/500#-N7W03YK0P9nSHAfPQRY:-N7XBjei3tJgvPG9Mv6j:b-896fix) ([raw file](https://github.com/pyomeca/bioptim/blob/1cc3c2477d261f4306d6f22b67b511108928ff6d/bioptim/optimization/solution.py#L1048)):*
<details><summary><i>Previously, pariterre (Pariterre) wrote…</i></summary><blockquote>

This looks like it is additive conditions (if .. AND ... AND...) 
If so, please make it explicit in a single if instead of nesting if

I think it also make sense to test (objective.target) first 
</blockquote></details>

Done.

File: bioptim/optimization/solution.py [hunk 1]
```diff
@@ -1042,11 +1042,11 @@ def animate(
...
             all_bioviz.append(bioviz.Viz(self.ocp.nlp[idx_phase].model.path().absolutePath().to_string(), **kwargs))
             all_bioviz[-1].load_movement(self.ocp.nlp[idx_phase].variable_mappings["q"].to_second.map(data["q"]))
+            cond = (ObjectiveFcn.Mayer.TRACK_MARKERS, ObjectiveFcn.Lagrange.TRACK_MARKERS, Node.ALL, Node.ALL_SHOOTING)
             for objective in self.ocp.nlp[idx_phase].J:
-                if objective.type in (ObjectiveFcn.Mayer.TRACK_MARKERS, ObjectiveFcn.Lagrange.TRACK_MARKERS):
-                    if objective.node[0] in (Node.ALL, Node.ALL_SHOOTING):
-                        if objective.target is not None:
-                            all_bioviz[-1].load_experimental_markers(objective.target[0])
+                if objective.target is not None:
+                    if objective.type in cond and objective.node[0] in cond:
```
[the remark this answers is not in the document]
Comment (thasaarah) [reply bioptim/optimization/solution.py]: Done

File: bioptim/limits/constraints.py [hunk 2]
```diff
@@ -533,7 +533,6 @@ def get_type() -> Callable
     CONTINUITY = (PenaltyFunctionAbstract.Functions.continuity,)
     TRACK_CONTROL = (PenaltyFunctionAbstract.Functions.minimize_controls,)
     TRACK_STATE = (PenaltyFunctionAbstract.Functions.minimize_states,)
-    TRACK_QDDOT = (PenaltyFunctionAbstract.Functions.minimize_qddot,)
```
Comment (Ipuch) [inline bioptim/limits/constraints.py:536]: Can you restore this enum?

**Review by pariterre — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pyomeca/bioptim/500#-:-N84rGJZ6MZs4kLlouNM:bnfp4nl)* status: 1 of 2 files reviewed, 1 unresolved discussion (waiting on @Ipuch and @pariterre)

**Pull request merged**